rfc/0010-rancher-integration-policies.md

```diff
@@ -0,0 +1,162 @@
+   it can be installed via Rancher Explorer UI.
+6. As a 3rd party developer, I want to release a policy on Artifact Hub so it can
+   be installed via Rancher Explorer UI.
+
+# Detailed design
+[design]: #detailed-design
+
+## For policies owned by the Kubewarden team
+
+Follow the process listed in [RFC-9, Rancher integration of Kubewarden
+charts](./0009-rancher-integration-charts.md).
+
+Carry the Helm chart code for the policy in a `rancher-X` orphan branch, where `X`
+is the release tag of the policy being targeted.
+Using these branches allows for easy rebases and cherry-picks, and documentating
+needed changes per commit. This also separates the Rancher vendored code, which
+may be of no interest to policy authors, and may coexist with other vendors.
+
+The resulting charts from `rancher-X` will not be served and released by us, but
+used to build Rancher charts from source by using the
+[Package](https://github.com/rancher/charts/blob/dev-v2.6/docs/packages.md)
+format in https://github.com/rancher/charts.
+They will be submitted via a `packages/kubewarden/<policy>.package.yaml`, with
+`package.yaml::commit` pointing to the relevant `rancher-X` branch.
+
+For policy bundles, they can be in their own repository containing the chart
+and modifications in a `rancher-X` branch (e.g: see `kubewarden-defaults`).
+
+## For 3rd party artifacts
+
+Use the provided [artifacthub.io public API](https://artifacthub.io/docs/api/).
+Currently, the Artifact Hub API provides [`GET
+/packages/search`](https://artifacthub.io/docs/api/#/Packages/searchPackages),
+where one can set the `repository kind` to `Kubewarden policies`. With this call
+one can obtain the list of Kubewarden policies. One would then query for each of
+the policies with [GET
+/packages/kubewarden/{repoName}/{packageName}](https://artifacthub.io/docs/api/#/Packages/getKubewardenPoliciesDetails),
+and obtain its `name`, `version`, `signed` status, and `data/kubewarden-*`
+information.
+
+Artifacts owned by the Kubewarden team are also present in Artifact Hub, hence
+to avoid listing them twice, they should be skipped if `provider` of the package
+is `kubewarden`, and/or the URL starts with `https://github.com/kubewarden/`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## For policies owned by the Kubewarden team
+
+Needing new Helm charts to ship policies owned by the Kubewarden team.
+
+A pull request bot needs to be created to make the submissions to
+https://github.com/rancher/charts, and we will need external approvals for
+merging.
+
+## For 3rd party policies
+
+Using the provided Artifact Hub API, given that it doesn't have pagination and
+one needs to query for each package independently, will be inefficient. A cache
+could be implemented.
+
+Obtaining the policies via the API is still missing:
+- Metadata not present in `policy/artifacthub-pkg.yml` yet present in
+  `policy/metadata.yml`, namely the array of `rules`.
```

Comment:
If we **really** want to provide a better integration with Rancher Manager, we could add all these information into the custom annotations that `artifacthub-pkg.yml` has. We could even provide a tool that takes `metadata.yml` and updates an existing `artifacthub-pkg.yml` file.

Reply:
True, it's the obvious play. I should have been more verbose there, fixed in last version.